.github/workflows/approve-docs.yml

```diff
@@ -0,0 +1,31 @@
+name: Auto-Approve Docs
+on: pull_request_target
+
+jobs:
+  review:
+    if: '${{ github.ref_protected }}'
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: 'Check scope of PR'
+        run: |
+          base="$GITHUB_BASE_REF"
+          branch="$GITHUB_HEAD_REF"
+          echo "Fetching branches"
+          git fetch origin "$base" "$branch"
+          echo
+          echo "Comparing $base..$branch"
+          dirs=$(./scripts/check-scope.sh "remotes/origin/$base" "remotes/origin/$branch")
+          echo
+          echo "Set scope output"
+          echo "::set-output name=scope::$dirs"
+        id: check
+
+      # Note that GitHub actions do not run from forked repos, so this
+      # will only approve PRs from branches in the main repo.
+      - name: 'Approve Docs'
+        if: "${{ steps.check.outputs.scope == 'docs' }}"
```

Comment:
The output of check-scope.sh is an array, which is then assigned to `steps.check.outputs.scope`. So you're checking that an array is equal to a single element? What happens in that case?

Reply:
It's not an array, just a normal string.
So I want to check that there was exactly one top-level directory changed, and it was `docs`. This can be done with string comparison.

Your point stands, however. I realise that I was not handling multi-line strings properly when setting the output variable. I just pushed an amended commit which reworks newline handling.